api/src/models/JobMetadata.ts

```diff
@@ -58,31 +60,61 @@ class JobMetadata {
     }
 
     get derivativeStatus(): Record<string, unknown> {
-        // TODO: populate with real data
+        let lockfileExists: boolean = fs.existsSync(this.derivativeLockfile);
         const status = {
             expected: 10,
             processed: 0,
             building: false,
+            lockfileExists
         };
+        this.order.pages.forEach(element => {
+            let image: Image = new Image(this.job.dir + "/" + this.page.filename )
+            Object.keys(image.sizes).forEach(key => {
+                status.expected += 1;
+                if (fs.existsSync(image.derivativePath(key))) {
+                    status.processed += 1;
+                }
+            })
+        });
         return status;
     }
 
     get uploadTime(): number {
-        // TODO: populate with real data
-        return 0;
-    }
+        this.order.pages.forEach(element => {
+            let getFileUpdatedDate = (path: string = this.job.dir + "/" + this.page.filename) => {
```

Comment:
I don't think this is doing what it needs to do. The original code is looking for the newest file within the job, so it iterates through files and keeps track of the newest time it has encountered so far. If it finds nothing at all, it falls back to the age of the directory itself. The forEach code doesn't seem to be doing any comparison to find a new date, and nothing seems to be manipulating the this.uploadTime property that the function ultimately returns (is that even defined anywhere?). I'm also not sure why there is a callback that returns a call to a callback here...